README.md

```diff
@@ -31,8 +45,161 @@ $ docker-compose up
+and reloading the data, but allow the candidate to suggest possible implementations.
+
+##### Evaluation criteria
+
+Does the candidate:
+
+1. Understand why the in-memory structure works, and why it is empty after a
+   restart?
+2. Understand basic I/O concepts?
+3. Discuss tradeoffs between the different options of when to save and load
+   data? Depending on their implementation, this could involve:
+   - Having two sources of truth in a file persisted data store and an in-memory store?
+   - Performance implications of reading the entire file on every request vs at start-up?
+4. Understand the life cycle of server: process re/start, request, response and where
+   in the framework they can respond to them?
+5. Handle empty cases (i.e., starting with no keys loaded)?
+6. Allow duplicate keys? The in-memory dictionary structure updates existing keys.
+
+#### 3. Fetch key by name
+
+Add a new endpoint that returns the value of the provided key.
+
+##### Evaluation criteria
+
+Does the candidate:
+
+1. Understand how to add a new route and function?
+2. Use sensible names for the path and the key?
+3. Demonstrate understanding of the methods for encoding the query in the
+   request? (i.e. query string param vs path argument vs form encoded POST)
+4. Handle common edge cases, like non-existing keys and null keys?
+5. Are they concerned about sanitizing user input?
+
+#### 4. Filter keys by prefix
+
+Add or extend an endpoint to return a list of all keys and values that match the
+provided prefix. For example, the prefix "na" would match "name" and "nationality".
+
+As an extension, have them sort the returned list in lexicographical key order.
+
+##### Evaluation criteria
+
+1. Is their method of filtering keys efficient?
+2. Does it handle edge cases as above, and do they consider case sensitivity?
+
+#### 5. Delete key by name
+
+Delete the entry for a key provided. This will likely be a new endpoint, but allow
+the candidate to suggest the implementation.
+
+##### Evaluation criteria
+
+1. This is similar to the fetch endpoint added above, did they retain knowledge
+   and improve the speed at which they could add this feature?
+
+#### Additional tasks
+
+In the unlikely event that the candidate makes it through all the previous tasks in
+the alloted time, ask the candidate to suggest other improvements they would make
+and engage in conversations about their implementation.
+
+## General evaluation criteria
+
+Here are additional high level considerations, taken from [The Software Engineer's Guide to
+Interviewing](https://blog.usejournal.com/the-software-engineers-guide-to-interviewing-software-engineers-980bbfdb4006):
+
+1. Can they write code?
+2. Can they make sensible design decisions?
+3. Can they work well with other people?
+
+### Common anti-patterns
+
+- Poor communicator
+- Poor code quality
+- Doesn't explore the problem before jumping into a solution
+- Broad gaps in knowledge
+- Not self-motivated
```

Comment:
How would you see this manifesting itself in an interview like this?

Reply:
I copied these bullets verbatim from the blog post, but absent the associated context, it may be better for me to take a pass at these to add more detail.